### PR TITLE
HLRC: adding machine learning delete job

### PR DESCRIPTION
**Pull request by benwtrent**
This adds the DELETE Job call for the Machine Learning client in the HLRC. 

Relates to #29827

### CODE REVIEW

**Comment by droberts195** [comment]
Did you try building the docs into HTML?  If not, this is what to do:

```
git clone git@github.com:elastic/docs.git
cd docs
./build_docs.pl --doc ~/elasticsearch/docs/java-rest/index.asciidoc
open html_docs/index.html
```

(If you haven't cloned the `elasticsearch` repo in your home directory then change `~` to wherever in the commands above.  If you get an error about certificates then edit `lib/ES/Util.pm` in the `docs` repo and add a `-k` argument to the `curl` invocation.)

Then navigate to your new documentation page and make sure it looks sensible in your browser.

File: docs/java-rest/high-level/supported-apis.asciidoc [hunk 1]
```diff
@@ -205,8 +205,10 @@ include::licensing/delete-license.asciidoc[]
 The Java High Level REST Client supports the following Machine Learning APIs:
 
 * <<java-rest-high-x-pack-ml-put-job>>
+* <<java-rest-high-x-pack-ml-delete-job>>
 
 include::ml/put_job.asciidoc[]
+include::ml/delete-job.asciidoc[]
```
[the remark this answers is not in the document]
Comment (benwtrent) [reply docs/java-rest/high-level/supported-apis.asciidoc]: Sounds good @droberts195

**Comment by benwtrent** [comment]
@droberts195 yes, I have built the docs and looked through them to make sure they look ok.

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
@benwtrent I think after merging this is a good time for me to refactor the request converters into their own class. It shouldn't take me long.

**Pull request merged**